types/src/proof/definition.rs

```diff
@@ -303,6 +304,62 @@ impl AccumulatorConsistencyProof {
     pub fn siblings(&self) -> &[HashValue] {
         &self.siblings
     }
+
+    /// Verifies the proof. The verification consists of two steps. First, the frozen subtree roots
+    /// should match `old_root_hash` and `old_num_leaves`. Second, after combining the frozen
+    /// subtree roots with the siblings we should be able to obtain `new_root_hash`.
+    pub fn verify<H: CryptoHasher>(
+        &self,
+        old_root_hash: HashValue,
+        old_num_leaves: u64,
+        new_root_hash: HashValue,
+        new_num_leaves: u64,
+    ) -> Result<()> {
+        ensure!(
+            old_num_leaves <= new_num_leaves,
+            "Old accumulator ({} leaves) is bigger than new accumulator ({} leaves).",
+            old_num_leaves,
+            new_num_leaves,
+        );
+
+        let old_accumulator =
+            Accumulator::<H>::new(self.frozen_subtree_roots.clone(), old_num_leaves)?;
+        ensure!(
+            old_accumulator.root_hash() == old_root_hash,
+            "Frozen subtree roots in the proof do not match old root hash. \
+             Actual root hash: {:x}. Expected root hash: {:x}.",
+            old_accumulator.root_hash(),
+            old_root_hash,
+        );
+
+        if old_num_leaves == 0 {
+            ensure!(
+                self.siblings.is_empty(),
+                "Proving consistency for empty accumulator does not need any siblings.",
+            );
+            return Ok(());
+        }
+
+        let expected_num_siblings = new_num_leaves.next_power_of_two().trailing_zeros() + 1
+            - old_num_leaves.count_ones()
```

Comment:
e.g. `// These are the frozen subtree nodes`  correct?   If so, why not count them directly from self.frozen_subtree_roots.len()

Reply:
There are not...

Reply:
Yeah they are the same.